src/components/schedule-page/validation-drawer/error-list-item.component.tsx

```diff
@@ -1,17 +1,59 @@
-import React from "react";
+import React, { useContext } from "react";
 import { Button } from "../../common-components";
 import { ScheduleErrorMessageModel } from "../../../common-models/schedule-error-message.model";
+import { ScheduleLogicContext } from "../table/schedule/use-schedule-state";
+import { TranslationHelper } from "../../../helpers/translations.helper";
+import { ApplicationStateModel } from "../../../state/models/application-state.model";
+import { useSelector } from "react-redux";
 
 interface Options {
   error: ScheduleErrorMessageModel;
 }
 
 export default function ErrorListItem({ error }: Options): JSX.Element {
+  /* eslint-disable @typescript-eslint/camelcase */
+  const { month_number } = useSelector(
+    (state: ApplicationStateModel) => state.scheduleData.present.schedule_info
+  );
+
+  let currMonthGenetivus = "";
+  let prevMonthGenetivus = "";
+  if (month_number) {
+    currMonthGenetivus = `${TranslationHelper.polishMonthsGenetivus[month_number]}`;
+    if (month_number > 0) {
+      prevMonthGenetivus = `${TranslationHelper.polishMonthsGenetivus[month_number - 1]}`;
+    } else {
+      prevMonthGenetivus = `${TranslationHelper.polishMonthsGenetivus[month_number + 11]}`;
+    }
+  }
+
+  const scheduleLogic = useContext(ScheduleLogicContext);
+  const mappedDays = scheduleLogic?.sections.Metadata?.verboseDates.map((d) => d.date);
+
+  let monthStartIndex = 0;
+  while (scheduleLogic?.sections.Metadata?.dates[monthStartIndex] !== 1) {
```

Comment:
Tutaj można to uprościć i użyć findIndex: https://www.w3schools.com/jsref/jsref_findindex.asp. Coś w tym stylu:
`  const monthStartIndex = scheduleLogic?.sections.Metadata?.dates.findIndex(d=> d === 1) ?? 0;`